homeassistant/components/hdmi_cec.py

```diff
@@ -30,92 +113,304 @@
+                                                    cv.string)
+                                            })),
+        vol.Optional(CONF_PLATFORM): vol.Any(SWITCH, MEDIA_PLAYER),
+        vol.Optional(CONF_HOST): cv.string,
     })
 }, extra=vol.ALLOW_EXTRA)
 
 
+def pad_physical_address(addr):
+    """Right-pad a physical address."""
+    return addr + [0] * (4 - len(addr))
+
+
 def parse_mapping(mapping, parents=None):
     """Parse configuration device mapping."""
     if parents is None:
         parents = []
     for addr, val in mapping.items():
-        cur = parents + [str(addr)]
-        if isinstance(val, dict):
-            yield from parse_mapping(val, cur)
-        elif isinstance(val, str):
-            yield (val, cur)
-
-
-def pad_physical_address(addr):
-    """Right-pad a physical address."""
-    return addr + ['0'] * (MAX_DEPTH - len(addr))
+        if isinstance(addr, (str,)) and isinstance(val, (str,)):
+            from pycec.network import PhysicalAddress
+            yield (addr, PhysicalAddress(val))
+        else:
+            cur = parents + [addr]
+            if isinstance(val, dict):
+                yield from parse_mapping(val, cur)
+            elif isinstance(val, str):
+                yield (val, pad_physical_address(cur))
 
 
-def setup(hass, config):
+def setup(hass: HomeAssistant, base_config):
     """Setup CEC capability."""
-    global _CEC
-
-    try:
-        import cec
-    except ImportError:
-        _LOGGER.error("libcec must be installed")
-        return False
+    from pycec.network import HDMINetwork
+    from pycec.commands import CecCommand, KeyReleaseCommand, KeyPressCommand
+    from pycec.const import KEY_VOLUME_UP, KEY_VOLUME_DOWN, KEY_MUTE, \
+        ADDR_AUDIOSYSTEM, ADDR_BROADCAST, ADDR_UNREGISTERED
+    from pycec.cec import CecAdapter
+    from pycec.tcp import TcpAdapter
 
     # Parse configuration into a dict of device name to physical address
     # represented as a list of four elements.
-    flat = {}
-    for pair in parse_mapping(config[DOMAIN].get(CONF_DEVICES, {})):
-        flat[pair[0]] = pad_physical_address(pair[1])
+    device_aliases = {}
+    devices = base_config[DOMAIN].get(CONF_DEVICES, {})
+    _LOGGER.debug("Parsing config %s", devices)
+    device_aliases.update(parse_mapping(devices))
+    _LOGGER.debug("Parsed devices: %s", device_aliases)
 
-    # Configure libcec.
-    cfg = cec.libcec_configuration()
-    cfg.strDeviceName = 'HASS'
-    cfg.bActivateSource = 0
-    cfg.bMonitorOnly = 1
-    cfg.clientVersion = cec.LIBCEC_VERSION_CURRENT
+    platform = base_config[DOMAIN].get(CONF_PLATFORM, SWITCH)
 
-    # Setup CEC adapter.
-    _CEC = cec.ICECAdapter.Create(cfg)
+    loop = (
+        # Create own thread if more than 1 CPU
+        hass.loop if multiprocessing.cpu_count() < 2 else None)
+    host = base_config[DOMAIN].get(CONF_HOST, None)
+    if host:
+        adapter = TcpAdapter(host, name="HASS", activate_source=False)
```

Comment:
Small suggestion: Can we use the full name "Home Assistant" rather than the abbreviation "HASS"?